src/history.cpp

```diff
@@ -1073,6 +1076,15 @@ void history_impl_t::clear() {
     this->clear_file_state();
 }
 
+void history_impl_t::clear_session() {
+    for (size_t i = 0; i < new_items.size(); i++) {
+        deleted_items.insert(new_items.at(i).str());
```

Comment:
nit: you could use a range-based for loop here to simplify it:

    for (const auto &item : new_items) {
        deleted_items.insert(item.str());
    }

Reply:
Changed this in later commit.